source/tutorials/nixos-configuration-on-vm.md

```diff
@@ -0,0 +1,187 @@
+<!-- todo links: ancors to introductions to library packages and modules are currently not in the manuals -->
+A NixOS configuration is a file that contains a function that is structured acording to the module<!-- todo link -->  convention.
+A NixOS configuration is a function that follows the convention of a module<!-- todo link to reference-->. <!-- remove one version -->
+
+<!-- [old version 2022.08.10 should be deleted] -->
+<!-- The Git repository Nixpkgs provides the necessary infrastructure and information needed to specify a NixOS configuration[^nixpkgs]. -->
+<!-- Nixpkgs contains a library, packages, and modules. -->
+<!-- The library provides the functional infrastructure to conveniently create packages. -->
+<!-- The packages defined in Nixpkgs provide build scripts needed to build software. -->
+<!-- The NixOS modules are a way to configure the system. [previous formulation] provide a configuration infrastructure to generate the system's configuration files. -->
+
+## Starting from default NixOS Configuration
+
+On NixOS, the configuration file is normally located at `/etc/nixos/configuration.nix`.
+A different location can be specified using the environment variable `$NIX_PATH`.
+For CLI commands that use a configuration, the file path can be specified.
+In this guide this strategy is preferred because the goal is not to install or update NixOS, but to debug a configuration.
+
+On NixOS, you can use the `nixos-generate-config` command to create a configuration file that contains some useful defaults and configuration suggestions [^nixosconf].
+You can create a NixOS configuration in your working directory:
+```shell-session
+nixos-generate-config --dir ./
+```
+
+In the working directory you will find two files: `configuration.nix` and `hardware-configuration.nix`.
+The `hardware-configuration.nix` file is specific to your current installation.
+This file is irrelevant for this tutorial because it has no effect inside a virtual machine.
+The `configuration.nix` file contains various suggestions useful for the initial setup of a desktop computer.
+Without the comments the configuration file contains the following content:
+```nix
+{ config, pkgs, ... }:
+{
+  imports =
+    [
+      ./hardware-configuration.nix
+    ];
+
+  boot.loader.systemd-boot.enable = true;
+  boot.loader.efi.canTouchEfiVariables = true;
+
+  services.xserver.enable = true;
+
+  services.xserver.displayManager.gdm.enable = true;
+  services.xserver.desktopManager.gnome.enable = true;
+
+  system.stateVersion = "22.05";
+}
+```
+
+In this tutorial we focus on testing NixOS configurations on a virtual machine, in these cases you can remove the `hardware-configuration.nix` import.
+
+To be able to log in, you must uncomment the section that specifies the user "alice", or add the following lines to the configuration inside the curly bracket that contain the configuration specification:
```

Comment:
```suggestion
To be able to log in, you must uncomment the section that specifies the user `alice`, or add the following lines to the configuration inside the curly bracket that contain the configuration specification:
```

Another point of contention, because I had the discussion with @Radvendii and @roberth recently (who are both welcome to chime in, because my perception may be badly skewed after so many iterations):
```suggestion
To be able to log in, you must uncomment the section that specifies the user `alice`, or add the following lines to the configuration inside the curly bracket that contain the system definition:
```

Reply:
I'd avoid both specification and definition, because those are "meta" words that tend to refer to something more abstract than what we're discussing 99% of the time.

To illustrate my point, here is a _definition_ of definition: a statement of the meaning of a word or word group or a sign or symbol.

The overloaded meaning of _declaration_ and especially _definition_ is unfortunate. Nowadays it's possible to have a successful career in software without interacting much with the C family and its jargon, so I'd like to stick to a more natural vocabulary.

In this instance specifically, I think we can stick to simple terminology, as the point of a tutorial is to go through the motions rather than explain everything. "Configuration" seems good enough, or "configuration in `nodes.machine`" if you think it'd be unclear which part of the test this is about.

Reply:
@roberth you mean like that?
```suggestion
To be able to log in, you must uncomment the section that specifies the user "alice", or add the following lines to the configuration inside the curly bracket that contain the configuration:
```